File: simpleMediaPlayer/app/src/main/java/com/daeseong/simplemediaplayer/getMusicList.java
package com.daeseong.simplemediaplayer;

import android.database.Cursor;
import android.provider.MediaStore;
import android.util.Log;
import com.mpatric.mp3agic.ID3v2;
import com.mpatric.mp3agic.Mp3File;
import com.mpatric.mp3agic.ID3v1;
import com.mpatric.mp3agic.ID3v24Tag;
import java.util.ArrayList;

public class getMusicList {

    private static final String TAG = getMusicList.class.getSimpleName();

    private ArrayList<MusicInfo> musicList  = new ArrayList<MusicInfo>();

    private Mp3File mp3File;
    private ID3v2 id3v2Tag;

    public ArrayList<MusicInfo> getData(){

        try {

            //음악 폴더 선택
            musicList.clear();
            Cursor cursor = MusicApplication.getInstance().getAppContext().getContentResolver().query(
                    MediaStore.Audio.Media.EXTERNAL_CONTENT_URI,
                    new String[] { MediaStore.Audio.AudioColumns.ARTIST, MediaStore.Audio.AudioColumns.TITLE, MediaStore.Audio.AudioColumns.DATA },
                    MediaStore.Audio.AudioColumns.IS_MUSIC + " > 0",
                    null,
                    null
            );

            int nIndex = cursor.getColumnIndex(MediaStore.Audio.AudioColumns.DATA);

            while(cursor.moveToNext()) {

                if (nIndex == -1) {
                    continue;
                }

                String sMusicPath = cursor.getString(nIndex);

                //music3 폴더만 가져온다
                if (!String_util.getLastFolderName(sMusicPath).equals("music2")) {
                    continue;
                }

                try {

                    String sTitle = "";
                    String sArtist = "";
                    String sSing = "";

                    mp3File = new Mp3File(sMusicPath);
                    if (mp3File != null){

                        if (mp3File.hasId3v2Tag()) {
                            id3v2Tag = mp3File.getId3v2Tag();
                            sTitle = id3v2Tag.getTitle();
                            sArtist = id3v2Tag.getArtist();

                            Log.e(TAG, id3v2Tag.getTrack() + id3v2Tag.getAlbum() + id3v2Tag.getYear() + id3v2Tag.getGenre() + id3v2Tag.getComment());

                        } else {
                            id3v2Tag = new ID3v24Tag();
                            if (mp3File.hasId3v1Tag()) {
                                ID3v1 id3v1Tag = mp3File.getId3v1Tag();
                                sTitle = id3v1Tag.getTitle();
                                sArtist = id3v1Tag.getArtist();

                                Log.e(TAG, id3v1Tag.getTrack() + id3v1Tag.getAlbum() + id3v1Tag.getYear() + id3v1Tag.getGenre() + id3v1Tag.getComment());
                            }
                        }

                        /*
                        int secs = (int)mp3File.getLengthInSeconds() % 60;
                        int mins = (int)mp3File.getLengthInSeconds() / 60;
                        String sPlaytime = String.format("%02d:%02d", mins, secs);
                        Log.e(TAG, "Playtime: " + sPlaytime);
                        */

                        if (sTitle == null && sArtist == null) {
                            sSing = String.format("%s", String_util.getFileName(sMusicPath));
                        } else {
                            sSing = String.format("%s - %s", sTitle, sArtist);
                        }
                        //Log.e(TAG, "sSing: " + sSing);

                        MusicInfo info = new MusicInfo();
                        info.setMusicPath(sMusicPath);
                        info.setMusicName(sSing);
                        musicList.add(info);
                    }

                }  catch (Exception ex){
                    Log.e(TAG, ex.getMessage().toString());
                }
            }

            cursor.close();

        } catch (Exception ex) {
            Log.e(TAG, ex.getMessage().toString());
        }

        return musicList;
    }
}

File: simpleMediaPlayer(kotlin)/app/src/main/java/com/daeseong/simplemediaplayer/getMusicList.kt
package com.daeseong.simplemediaplayer

import android.database.Cursor
import android.provider.MediaStore
import android.util.Log
import com.mpatric.mp3agic.ID3v1
import com.mpatric.mp3agic.ID3v2
import com.mpatric.mp3agic.ID3v24Tag
import com.mpatric.mp3agic.Mp3File

class getMusicList {

    private val tag = getMusicList::class.java.simpleName

    private val musicList = ArrayList<MusicInfo>()

    private var mp3File: Mp3File? = null
    private var id3v2Tag: ID3v2? = null

    fun getData(): ArrayList<MusicInfo> {

        try {

            //음악 폴더 선택
            musicList.clear()
            val cursor: Cursor? = MusicApplication.getAppContext().contentResolver.query(
                MediaStore.Audio.Media.EXTERNAL_CONTENT_URI,
                arrayOf(MediaStore.Audio.AudioColumns.ARTIST, MediaStore.Audio.AudioColumns.TITLE, MediaStore.Audio.AudioColumns.DATA),
                "${MediaStore.Audio.AudioColumns.IS_MUSIC} > 0",
                null,
                null
            )

            val nIndex: Int = cursor?.getColumnIndex(MediaStore.Audio.AudioColumns.DATA) ?: -1

            cursor?.use {

                while (it.moveToNext()) {

                    if (nIndex == -1) {
                        continue
                    }

                    val sMusicPath: String = it.getString(nIndex)

                    //music3 폴더만 가져온다
                    if (String_util.getLastFolderName(sMusicPath) != "music2") {
                        continue
                    }

                    try {
                        var sTitle = ""
                        var sArtist = ""
                        var sSing = ""

                        mp3File = Mp3File(sMusicPath)
                        if (mp3File != null) {

                            if (mp3File?.hasId3v2Tag() == true) {
                                id3v2Tag = mp3File?.id3v2Tag
                                sTitle = id3v2Tag?.title ?: ""
                                sArtist = id3v2Tag?.artist ?: ""

                                Log.e(tag, "${id3v2Tag?.track}${id3v2Tag?.album}${id3v2Tag?.year}${id3v2Tag?.genre}${id3v2Tag?.comment}")

                            } else {
                                id3v2Tag = ID3v24Tag()
                                if (mp3File?.hasId3v1Tag() == true) {
                                    val id3v1Tag: ID3v1? = mp3File?.id3v1Tag
                                    sTitle = id3v1Tag?.title ?: ""
                                    sArtist = id3v1Tag?.artist ?: ""

                                    Log.e(tag, "${id3v1Tag?.track}${id3v1Tag?.album}${id3v1Tag?.year}${id3v1Tag?.genre}${id3v1Tag?.comment}")
                                }
                            }

                            sSing = if (sTitle == "" && sArtist == "") {
                                String.format("%s", String_util.getFileName(sMusicPath))
                            } else {
                                String.format("%s - %s", sTitle, sArtist)
                            }

                            val info = MusicInfo()
                            info.musicPath = sMusicPath
                            info.musicName = sSing
                            musicList.add(info)
                        }

                    } catch (ex: Exception) {
                        Log.e(tag, ex.message.toString())
                    }
                }
            }

        } catch (ex: Exception) {
            Log.e(tag, ex.message.toString())
        }

        return musicList
    }
}
